site/src/components/RequireAuth/RequireAuth.tsx

```diff
@@ -13,9 +13,10 @@ export const RequireAuth: React.FC<RequireAuthProps> = ({ children }) => {
   const xServices = useContext(XServiceContext)
   const [authState] = useActor(xServices.authXService)
   const location = useLocation()
-  const navigateTo = location.pathname === "/" ? "/login" : embedRedirect(location.pathname)
+  const isHomePage = location.pathname === "/"
+  const navigateTo = isHomePage ? "/login" : embedRedirect(location.pathname)
   if (authState.matches("signedOut")) {
-    return <Navigate to={navigateTo} />
+    return <Navigate to={navigateTo} state={{ isRedirect: !isHomePage }} />
```

Comment:
does `isRedirect` replicate the functionality of `embedRedirect`?

Reply:
I think we *could* use one both to know whether to show the error and where to go next, but it might be better to keep them separate like this

Reply:
Yeah, we’d probably have to retrieve the redirect in the component but that didn’t seem like a very solid approach.